### PR TITLE
cli: fixup service test delete by using atomic actions.

### PR DESCRIPTION
**Pull request by jrasell**
I believe the service delete test was suffering due to the eventual consistency between job registration, eval completion, allocation placement, allocation start, and service registration.

I have therefore changed the test to upsert the service registrations directly into state, avoiding this while still allowing the test to serve its purpose of testing the delete CLI. A nice secondary outcome is that it now also tests cross-namespace deletion as the mock generates two service registrations, within different namespaces.

### CODE REVIEW

**Review by tgross — APPROVED** [review]
:+1:

File: command/service_delete_test.go [hunk 1]
```diff
@@ -64,6 +64,7 @@ func TestServiceDeleteCommand_Run(t *testing.T) {
 	require.NoError(t, err)
 	registerCode := waitForSuccess(ui, client, fullId, t, regResp.EvalID)
 	require.Equal(t, 0, registerCode)
+	require.Equal(t, 0, waitForSuccess(ui, client, fullId, t, regResp.EvalID))
```
Comment (shoenig) [inline command/service_delete_test.go:67]: Isn't this exactly the same as the two lines above it?

Comment (jrasell) [reply command/service_delete_test.go]: Ohhh yeah! Sorry about that, tried to rush.

**Review by shoenig — APPROVED** [review]
LGTM

**Pull request merged**